### ISSUE TITLE
Bad prediction results

### ISSUE THREAD

**Issue by deepseek** [mrluin/ESFNet-Pytorch#11]
@mrluin 
After training the `WHU Building Dataset (The cropped aerial image tiles and raster labels)` for 300 epoch, I used the `checkpoint-best.pth` for predicting an image, but:
- The prediction results were really bad.
- Every time I ran `predict.py`, the predicted result was different from before each time.

The files are attached below, containing `(log, input, output, groundtruth, model)`:
[results.zip](https://github.com/mrluin/ESFNet-Pytorch/files/3256040/results.zip)

Here are some images of the results,
First example:

![result](https://user-images.githubusercontent.com/49877135/58939111-1423d600-877f-11e9-87eb-cb9127d0023b.jpg)

Second example:

![result2](https://user-images.githubusercontent.com/49877135/58939181-4c2b1900-877f-11e9-90fb-6de9a1740273.jpg)

**Comment by mrluin** [comment]
I am sorry that my checkpoint file is lost.
I will train it from scratch and upload the checkpoint-best.pth.

**Comment by deepseek** [comment]
I updated to the latest commit that you posted, and then I retrained, but when I run `predict.py` the results are still really bad, also they are inconsistent and change everytime I run `predict.py`.

**I will be waiting for you to upload your trained weights.**

**Comment by mrluin** [comment]
@deepseek 
1. I checked out that there are some bugs in `predict.py` and I have fixed it.
2. I have added configurations for duplicating the predictions in `predict.py`.
3. Uploaded my trained weight, but the result is slightly lower (depends on some non-deterministic operations).

**Comment by mrluin** [comment]
I think there must be something I have forgotten that causes this little difference. I tried my best to find the checkpoint file, but the time is too long :/

**Comment by deepseek** [comment]
Thanks, now I am getting better results.
example:

![6](https://user-images.githubusercontent.com/49877135/59522344-1b618700-8ed7-11e9-92e2-c204c273b8dd.png)

**Comment by zhaozhiyi11** [comment]
> I am sorry that my checkpoint file is lost. I will train it from scratch and upload the checkpoint-best.pth.

我和你遇到了同样的问题，请问您是怎么解决的？

**Comment by zhaozhiyi11** [comment]
> Thanks, now I am getting better results. example:
> 
> ![6](https://user-images.githubusercontent.com/49877135/59522344-1b618700-8ed7-11e9-92e2-c204c273b8dd.png)

我遇到了同样的问题，预测出来的结果和你一样，请问你是怎么解决的？
